ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -310,6 +310,9 @@ func (r *RayClusterReconciler) reconcileServices(instance *rayiov1alpha1.RayClus
 			if val, ok := instance.Spec.HeadGroupSpec.Template.ObjectMeta.Labels[common.KubernetesApplicationNameLabelKey]; ok {
 				labels[common.KubernetesApplicationNameLabelKey] = val
 			}
+			for k, v := range instance.Spec.HeadServiceCustomLabels {
```

Comment:
This may overwrite L311 `labels[common.KubernetesApplicationNameLabelKey] = val`. It may cause a label selector mismatch for the head Pod and head service. See https://github.com/ray-project/kuberay/pull/572 for more details.

For example, you can add a label in `HeadServiceCustomLabels` with key `common.KubernetesApplicationNameLabelKey` (i.e. "app.kubernetes.io/name") and a random value.

Reply:
Thank you for pointing this out, I have modified it accordingly.